pyboolnet/control_strategies.py

```diff
@@ -0,0 +1,486 @@
+        * *update*: type of update, either *"synchronous"*, *"asynchronous"* or *"mixed"*.
+
+    **returns**:
+        * True if the query is true, false otherwise.
+    """
+
+    target_vars = list(set([item for subs in target for item in subs]))
+    new_primes = fix_components_and_reduce(primes, subspace, keep_vars=target_vars)
+    return run_control_query(new_primes, target, update)
+
+
+def run_control_query(primes, target, update):
+    """
+    Runs the model checking query for control.
+
+    **arguments**:
+        * *primes*: prime implicants.
+        * *target*: list of subspaces defining the target subset.
+        * *update*: type of update, either *"synchronous"*, *"asynchronous"* or *"mixed"*.
+
+    **returns**:
+        * True if the query is true, False otherwise.
+    """
+
+    spec = 'CTLSPEC ' + EFAG_set_of_subspaces(primes, target)
+    init = "INIT TRUE"
+    return model_checking(primes, update, init, spec)
+
+
+def control_direct_percolation(primes, candidate, target, silent=False):
+    """
+    Checks whether the subspace *candidate* is a control strategy for *target* by direct percolation.
+
+    **arguments**:
+        * *primes*: prime implicants.
+        * *candidate*: subspace.
+        * *target*: list of subspaces defining the target subset.
+        * *silent*: if True, it does not print infos to screen. Default value: False.
+
+    **returns**:
+        * True if the *candidate* percolates into the *target*, False otherwise.
+    """
+
+    perc = find_constants(primes=percolate(primes=primes, add_constants=candidate, copy=True))
+    if any(is_included_in_subspace(perc, subs) for subs in target):
+        if not silent:
+            print("Intervention (only percolation):", candidate)
+            print("Percolation:", perc)
+        return True
+    return False
+
+
+def control_completeness(primes: dict, candidate: dict, target: dict, update: str, silent=False):
+    """
+    Checks whether the subspace *candidate* is a control strategy for *target* by completeness.
+
+    **arguments**:
+        * *primes*: prime implicants.
+        * *candidate*: subspace.
+        * *target*: subspace defining the target subset.
+        * *update*: type of update, either *"synchronous"*, *"asynchronous"* or *"mixed"*.
+        * *silent*: if True, it does not print infos to screen. Default value: False.
+
+    **returns**:
+        * True if the *candidate* is a control strategy by completeness for *target*, False otherwise.
+    """
+
+    if type(target) == list:
+        print("The target must be a subspace.")
+        return
+
+    perc = find_constants(primes=percolate(primes=primes, add_constants=candidate, copy=True))
+    new_primes = fix_components_and_reduce(primes, perc, keep_vars=list(target.keys()))
+    tsmin = compute_trap_spaces(new_primes, "min")
+
+    if not all(is_included_in_subspace(T, target) for T in tsmin):
+        return False
+
+    if completeness(new_primes, update):
```

Comment:
i strongly recommend to use named parameters instead of parameters by position.

the reason is that
- when the signature of `completeness` changes, this call might break
- it may be easier to read